indexer/indexer_tests/osm_editor_test.cpp

```diff
@@ -0,0 +1,558 @@
+#include "indexer/indexer_tests/osm_editor_test.hpp"
+#include "testing/testing.hpp"
+
+#include "search/reverse_geocoder.hpp"
+
+#include "indexer/classificator_loader.hpp"
+
+#include "indexer/classificator.hpp"
+#include "indexer/ftypes_matcher.hpp"
+#include "indexer/osm_editor.hpp"
+#include "indexer/index_helpers.hpp"
+
+#include "editor/editor_storage.hpp"
+
+using namespace generator::tests_support;
+
+namespace
+{
+class IsCafeChecker : public ftypes::BaseChecker
+{
+public:
+  static IsCafeChecker const & Instance()
+  {
+    static const IsCafeChecker instance;
+    return instance;
+  }
+
+private:
+  IsCafeChecker()
+  {
+    Classificator const & c = classif();
+    m_types.push_back(c.GetTypeByPath({"amenity", "cafe"}));
+  }
+};
+
+class TestCafe : public TestPOI
+{
+public:
+  TestCafe(m2::PointD const & center, string const & name, string const & lang)
+    : TestPOI(center, name, lang)
+  {
+    SetTypes({{"amenity", "cafe"}});
+  }
+};
+
+template <typename TFn>
+void ForEachCafeAtPoint(Index & index, m2::PointD const & mercator, TFn && fn)
+{
+  m2::RectD const rect = MercatorBounds::RectByCenterXYAndSizeInMeters(mercator, 0.2 /* rect width */);
+
+  auto const f = [&fn](FeatureType & ft)
+  {
+    if (IsCafeChecker::Instance()(ft))
+    {
+      fn(ft);
+    }
+  };
+
+  index.ForEachInRect(f, rect, scales::GetUpperScale());
+}
+
+void FillEditableMapObject(osm::Editor const & editor, FeatureType const & ft, osm::EditableMapObject & emo)
+{
+  emo.SetFromFeatureType(ft);
+  emo.SetHouseNumber(ft.GetHouseNumber());
+  emo.SetEditableProperties(editor.GetEditableProperties(ft));
+}
+}  // namespace
+
+namespace editor
+{
+namespace testing
+{
+EditorTest::EditorTest()
+{
+  try
+  {
+    classificator::Load();
+  }
+  catch (RootException const & e)
+  {
+    LOG(LERROR, ("Classificator read error: ", e.what()));
+  }
+
+  InitEditorForTest();
```

Comment:
This function is called once, in ctor. Please, inline its code here, and remove it then.